### ISSUE TITLE
Modal title and label missing warning

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly-react#4218]
**Describe the issue. What is the expected and unexpected behavior?**
Recently in the [wizard refactor](https://github.com/patternfly/patternfly-react/pull/4140), we made the title prop for modal optional. This change makes sense for a lot of reasons, including being able to reference the wizard title as the aria-label or title for the modal and allows more flexibility in usage. Also, as Coker describes in this [issue](https://github.com/patternfly/patternfly-react/issues/3840): It used to be that if you omitted title, you ended up with an empty `<h1 class="pf-c-title">` element, which wasn't ideal from a code perspective, and broke the layout by creating extra space at the top of the modal from the empty element. To get the proper layout in a modal without a title, you used to have to specify the hideTitle prop which seems unnecessary. The modal doesn’t render an empty `<h1>` if you omit the title now which makes `hideTitle` unnecessary.

However, there is an accessibility concern if a consumer chooses to omit title and aria-label. Labeling is the most common accessibility issue that I've seen, as it's very easy to forget or not consider labeling. If they have neither attribute in a modal, it gives the user very little context that they’re inside of a modal. In a basic modal without either attribute, VO will read “close button” as soon as the modal opens and without any titles or aria-labels with the wizard in the modal, focus would go to the next button and read "steps, dialog, next, button, footer". Both of which would make the modal a bad accessible experience.

Modal needs some sort of title in it  - whether it’s the proper modal title (title prop), or you use the noModalBodyWrapper and you have an `<h1>` or whatever in it. If there isn’t a header, the modal needs either `aria-label` or `aria-labelledby` that points to a title that could serve as the label for the modal. It would be ideal if we could provide a warning to consumers if they don’t provide a title or an aria-label or if there was some way to notify users of the potential issue. We have done something similar in table: https://github.com/patternfly/patternfly-react/blob/v4/packages/react-table/src/components/Table/Table.tsx#L384-L387 . 

This would work for the most part, except in the case of the wizard or just a use of noModalBodyWrapper where the user has specified their own title in the modal. We have no way of knowing whether they supplied a title in the modal or not, so if they do have a title in their modal, but didn’t use our title or `aria-label` or `aria-labelledby`, we would end up giving them a warning even though they have a title and are using the modal correctly. 

The question is how to warn or notify users of potentially serious accessibility issues. We want to allow flexibility in our components but also ensure that we're not encouraging more accessibility errors in product usage.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Enhancement to prevent accessibility issues for consumers

**Comment by seanforyou23** [comment]
Closed with https://github.com/patternfly/patternfly-react/pull/4239
